lib/puppet/configurer.rb

```diff
@@ -49,6 +51,14 @@ def initialize(factory = Puppet::Configurer::DownloaderFactory.new)
     @splayed = false
     @environment = Puppet[:environment]
     @transaction_uuid = SecureRandom.uuid
+    @static_catalog = true
+
+    # Get checksum types known to the agent, preferring the default specified
+    # in the digest_algorithm setting.
+    @checksum_type = valid_checksum_types
```

Comment:
:+1: to excluding things other than `md5` and `sha256`.

One problem with sending multiple checksums is we don't have a way for a FIPS-enabled agent to say never use `md5`. For example, with this PR, you could set `Puppet[:digest_algorithm] = 'sha256'` on the agent, but the agent would still send `sha256,md5`, which would be misleading. It seems like we either need to send a single checksum type, or allow the list of checksum types to be configurable.